### PR TITLE
Simplify abort task and refactor recipe validation errors

### LINKED ISSUE #859: Server crash trying to generate AbortJob
If the server can't parse the recipe (e.g. if the wrong format is sent from an older client, #858), then the server will try to generate an AbortTask.  This is currently failing though:
```
2021-08-04 20:38:55 [INFO] Turbinia version: 20210611.dev20+gdd6b969
2021-08-04 20:38:57 [INFO] Disabling non-allowlisted jobs configured to be disabled in the config file: volatilityjob, binaryextractorjob, bulkextractorjob, photorecjob
2021-08-04 20:38:57 [INFO] Starting Prometheus endpoint.
2021-08-04 20:38:57 [INFO] Running Turbinia Server.
2021-08-04 20:38:57 [INFO] Starting Task Manager run loop
2021-08-04 20:49:57 [INFO] Processing PubSub message 2818205155225310
2021-08-04 20:49:57 [WARNING] No globals recipe specified, all recipes should include a globals entry, the default values will be used
2021-08-04 20:49:57 [ERROR] Recipe item jobs_allowlist has no "task" key. All recipe items must have a "task" key indicating the TurbiniaTask to which it relates.
2021-08-04 20:49:57 [INFO] Received evidence [GoogleCloudDisk:turbinia-test-persistent-disk:None] from PubSub message.
2021-08-04 20:49:57 [INFO] Adding new evidence: GoogleCloudDisk:turbinia-test-persistent-disk:None
2021-08-04 20:49:57 [INFO] Writing new task AbortTask into Datastore
2021-08-04 20:49:57 [INFO] Recipe Invalid
2021-08-04 20:49:57 [INFO] Starting post-processor for evidence turbinia-test-persistent-disk
Traceback (most recent call last):
  File "/usr/local/bin/turbiniactl", line 11, in <module>
    load_entry_point('turbinia==20210611.dev20+gdd6b969', 'console_scripts', 'turbiniactl')()
  File "/usr/local/lib/python3.8/dist-packages/turbinia-20210611.dev20+gdd6b969-py3.8.egg/turbinia/turbiniactl.py", line 660, in main
    server.start()
  File "/usr/local/lib/python3.8/dist-packages/turbinia-20210611.dev20+gdd6b969-py3.8.egg/turbinia/server.py", line 64, in start
    self.task_manager.run()
  File "/usr/local/lib/python3.8/dist-packages/turbinia-20210611.dev20+gdd6b969-py3.8.egg/turbinia/task_manager.py", line 544, in run
    for task in self.process_tasks():
  File "/usr/local/lib/python3.8/dist-packages/turbinia-20210611.dev20+gdd6b969-py3.8.egg/turbinia/task_manager.py", line 698, in process_tasks
    elif psq_task.status not in (psq.task.FINISHED, psq.task.FAILED):
AttributeError: 'str' object has no attribute 'status'
Program shutting down, attempting to send 2 queued log entries to Cloud Logging...
Waiting up to 5 seconds.
Sent all pending logs.
```

### PR DESCRIPTION
**Pull request by aarontp**
* We don't actually need to create a full Job or the run method of the Task, so we can pull out the job object and most of the task methods and mocks, etc. 
* This also plumbs the recipe validation failure messages through so that the client gets more details on what failed.
* A new `abort_request()` method is created to handle the AbortTask creation
* The `abort_request()` gets called directly from `get_evidence()` where the recipe parsing happens so we don't need to plumb the abort messages through the evidence object and can remove the error handling from `add_evidence()`.
* The requester parameter moves into the 'globals' section of the recipe to match the other global parameters.
* Small unrelated change, but adding the --unattended flag to Plaso

Fixes #859

### CODE REVIEW

**Review by wajihyassine — COMMENTED** [review]
LGTM other than the nitpick in the plaso module

File: turbinia/workers/plaso.py [hunk 1]
```diff
@@ -127,7 +132,12 @@ def run(self, evidence, result):
 
     cmd.extend(['--temporary_directory', self.tmp_dir])
     cmd.extend(['--logfile', plaso_log])
+    cmd.extend(['--unattended'])
     cmd.extend([plaso_file, evidence.local_path])
+
+    self.result.log('DEBUUUUUG: command is {0!s}'.format(cmd))
```
[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/workers/plaso.py:138]: Oh, lol, oops!  Those weren't meant to stay in there.  I was using those to debug a separate issue.  Thanks for flagging that :).

[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/workers/plaso.py]: Looks like I had forgotten to push my last change to remove these, but they're gone now.

**Review by wajihyassine — APPROVED** [review]
LGTM!
